### DISCUSSION TITLE
Notification on MacOS Monterey 12.6

### DISCUSSION THREAD

**Discussion by Blatman** [tauri-apps/tauri#5340] [Q&A]
Trying the notification API and it works in dev and debug production but not in normal production.

Also even though it works in dev/debug prod the icon is always the Apple Terminal icon - I would have expected the app icon.

Looking through the bug history I could only find an issue where it only worked in production but not dev - I'm seeing the reverse.    

The permissions for the app look to be OK in the System Preferences>Notifications

Does the app icon need to be specified in a particular way and if it isn't will this produce the non-operation in production?

I can see in the notification.rs file a reference to &self.identifier which needs to be obtained and the lack of this may be why the icon is always the Terminal icon.

JS -

function notify(content) {
	    var payload = JSON.parse(content);
        notificationEvent.sendNotification({ title: payload.title, body: payload.message, icon: payload.icon });
    }

    // this needs to be down here - further up will stop main window appearing for some reason ???
    let permissionGranted = await notificationEvent.isPermissionGranted();

    if (!permissionGranted) {
        const permission = await notificationEvent.requestPermission();
        permissionGranted = permission === 'granted';
    }
    if (permissionGranted) {
	    notificationEvent.sendNotification({ title: 'Tauri', body: 'Tauri is awesome!', icon: 'icon' });  // always shows Terminal icon???
    }

**Comment by FabianLars** [comment]
I don't know why it doesn't work in production, but in development the Terminal Icon is expected (because macOS is a little weird - you can't set the icon manually, only the application which doesn't really exist in dev).

Can you add the `"devtools"` feature flag to the tauri dependency in Cargo.toml and check the devtools console for errors in production?

**Reply by Blatman** [reply]
OK thanks for that - I have enabled the devtools in Cargo.toml (very handy) but all logging is OK however the notifications did start to work for a wee bit but then they broke again - still trying to figure it out but not sure if it was something to do with checking the System Preference settings.
BTW - when they did work the icon displayed was the app icon so just need to try and find out why they are coming and going.

**Reply by Blatman** [reply]
OK - it seems that a lot of folk are missing notifications since upgrading to Monterey and a few have isolated it to the `when mirroring/sharing display` setting in System Prefs particularly if apps like Zoom are in use. In my case I found the Alerts appeared to work but Banners were intermittent however after checking the mirroring/sharing setting the banners seem to work OK (I don't have Zoom but have Signal/Skype) so I think it is an OS issue and as pointed out above the icon will work OK in Prod.

**Reply by Blatman** [reply]
Yup - Banners definitely flakey but Alerts are rock solid - still can't quite put my finger on why the Banners don't work properly.

Tried a Universal build on 10.13 and both Banners/Alerts work OK.

Just went back to Monterey and stumbled on something - if I have the app main window focussed the Banners will not work but if I focus on the Notifications/Focus panel of Systems Preferences the Banners from the app work every time. I am triggering the notifications from a System Tray item so the focus can be shifted to test. 
So now my head is mince, I'm going to bed!
